api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/jena/JenaAdminActions.java

```diff
@@ -192,29 +192,33 @@ private String testWriteXML() {
 	}
 
     private void printRestrictions() {
-    	TBoxReasonerModule reasoner = ApplicationUtils.instance().getTBoxReasonerModule();
-    	for (Restriction rest : reasoner.listRestrictions() ) {
-    		if (rest.isAllValuesFromRestriction()) {
-    			log.trace("All values from: ");
-    			AllValuesFromRestriction avfr = rest.asAllValuesFromRestriction();
-    			Resource res = avfr.getAllValuesFrom();
-    			if (res.canAs(OntClass.class)) {
-    				OntClass resClass = res.as(OntClass.class);
-    				for (Resource inst : resClass.listInstances().toList() ) {
-    					log.trace("    -"+inst.getURI());
-    				}
-    			}
-    		} else if (rest.isSomeValuesFromRestriction()) {
-    			log.trace("Some values from: ");
-    		} else if (rest.isHasValueRestriction()) {
-    			log.trace("Has value: ");
-    		}
-    		log.trace("On property "+rest.getOnProperty().getURI());
-    		for (Resource inst : rest.listInstances().toList() ) {
-    			log.trace("     "+inst.getURI());
-    		}
-
-    	}
+		try {
+			TBoxReasonerModule reasoner = ApplicationUtils.instance().getTBoxReasonerModule();
+			for (Restriction rest : reasoner.listRestrictions() ) {
+				if (rest.isAllValuesFromRestriction()) {
+					log.trace("All values from: ");
+					AllValuesFromRestriction avfr = rest.asAllValuesFromRestriction();
+					Resource res = avfr.getAllValuesFrom();
+					if (res.canAs(OntClass.class)) {
+						OntClass resClass = res.as(OntClass.class);
+						for (Resource inst : resClass.listInstances().toList() ) {
+							log.trace("    -"+inst.getURI());
+						}
+					}
+				} else if (rest.isSomeValuesFromRestriction()) {
+					log.trace("Some values from: ");
+				} else if (rest.isHasValueRestriction()) {
+					log.trace("Has value: ");
+				}
+				log.trace("On property "+rest.getOnProperty().getURI());
+				for (Resource inst : rest.listInstances().toList() ) {
+					log.trace("     "+inst.getURI());
+				}
+			}
+		}
+		catch (IllegalStateException e) {
+			log.debug("Status of reasoner could not be determined. It is likely disabled", e);
```

Comment:
Suggest changing the logging level to `warn`.

Reply:
Ok, I agree this one deserves a warning since the function doesn't work at all if the reasoner is disabled.